lib/debug.ml

```diff
@@ -122,26 +122,28 @@ let format include_time brand priority message =
 let print_debug = ref false
 let log_to_stdout () = print_debug := true
 
+module BrandLevelPair = struct
+  type t = string * Syslog.level
+  let compare = Pervasives.compare
+end
+module BrandLevelPairSet = Set.Make(BrandLevelPair)
+
 let loglevel_m = Mutex.create ()
-let logging_disabled_for : (string * Syslog.level, unit) Hashtbl.t = Hashtbl.create 0
+let logging_disabled_for = ref BrandLevelPairSet.empty
 let default_loglevel = Syslog.Debug
 let loglevel = ref default_loglevel
 
 let disabled_modules () =
-  Mutex.execute loglevel_m (fun () ->
-      Hashtbl.fold (fun key _ acc -> key :: acc) logging_disabled_for []
-    )
+    BrandLevelPairSet.fold (fun brandlevel acc -> brandlevel :: acc) !logging_disabled_for []
```

Comment:
Is this the same as `BrandLevelPairSet.elements !logging_disabled_for`?

Reply:
Thanks I've updated it